services/lambda/src/it/java/software/amazon/awssdk/services/lambda/IntegrationTestBase.java

```diff
@@ -91,24 +91,45 @@ public static void tearDown() {
     }
 
     private static File setupFunctionZip(String jsFile) throws IOException {
-        InputStream in = IntegrationTestBase.class.getResourceAsStream(jsFile);
+        try {
 
-        File zipFile = File.createTempFile("lambda-cloud-function", ".zip");
-        ZipOutputStream out = new ZipOutputStream(new FileOutputStream(zipFile));
+            InputStream in = IntegrationTestBase.class.getResourceAsStream(jsFile);
 
-        out.putNextEntry(new ZipEntry(jsFile));
+            File zipFile = File.createTempFile("lambda-cloud-function", ".zip");
+            ZipOutputStream out = new ZipOutputStream(new FileOutputStream(zipFile));
 
-        byte[] b = new byte[1024];
-        int count;
+            out.putNextEntry(new ZipEntry(jsFile));
 
-        while ((count = in.read(b)) != -1) {
-            out.write(b, 0, count);
-        }
+            byte[] b = new byte[1024];
+            int count;
 
-        out.close();
-        in.close();
+            while ((count = in.read(b)) != -1) {
+                out.write(b, 0, count);
+            }
 
-        return zipFile;
+            out.close();
+            in.close();
+            return zipFile;
+        }
+        catch(IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        finally {
+            if (in != null) {
+                try {
+                    in.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 118:124, 125:131.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.